src/pykeen/metrics/ranking.py

```diff
@@ -111,6 +114,30 @@ def generate_num_candidates_and_ranks(
     return ranks, num_candidates
 
 
+def weighted_mean_expectation(individual: np.ndarray, weights: Optional[np.ndarray]) -> float:
```

Comment:
move utility functions into `pykeen.utils`

Reply:
moved to `pykeen.metrics.utils`, cf. e12dd97...db0ce01,  since it is currently only used for metrics